### PR TITLE
Create module for Semtypes

### LINKED ISSUE #32224: Create a Gradle submodule for semtype
Create a Gradle sub module for semtype java code.
Create source and test directory.
Add a minimal test case and corresponding implementation, make the java code be one-to-one translation of ballerina code in nballerina repo.

### PR DESCRIPTION
**Pull request by ushirask**
Fixes #32224 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: semtypes/spotbugs-exclude.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!--
+  ~  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (rdhananjaya) [inline semtypes/spotbugs-exclude.xml:2]:
```suggestion
  ~  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: semtypes/spotbugs-exclude.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  ~  software distributed under the License is distributed on an
+  ~  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~  KIND, either express or implied.  See the License for the
+  ~  specific language governing permissions and limitations
+  ~  under the License.
+  -->
+
+<FindBugsFilter>
+    <Match>
+        <Package name="org.ballerina.semtypes"/>
```
Comment (manuranga) [inline semtypes/spotbugs-exclude.xml:21]: What is the name in nballerina side, semtype or semtypes

File: semtypes/src/main/java/org/ballerina/semtypes/SemType.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package org.ballerina.semtypes;
```
Comment (manuranga) [inline semtypes/src/main/java/org/ballerina/semtypes/SemType.java:1]: Lets match the file names, and check if upper case `T` in file name is correct .

File: semtypes/src/main/java/org/ballerina/semtypes/SemType.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * Contains core functions for implementing semantic typing.
+ *
+ */
+class SemtypePlaceholder {
```
Comment (manuranga) [inline semtypes/src/main/java/org/ballerina/semtypes/SemType.java:7]: What does this correspond to in nballerina side?

Comment (manuranga) [reply semtypes/src/main/java/org/ballerina/semtypes/SemType.java:7]: I dont think we need this

File: semtypes/spotbugs-exclude.xml (hunk 2 above)
Comment (ushirask) [reply semtypes/spotbugs-exclude.xml:21]: The type name is SemType in the types module

File: semtypes/src/test/java/org/ballerina/semtypes/SemtypeTests.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Tests if placeholder is created.
+ *
+ */
+public class SemtypeTests {
+
+    @Test
+    public void testPlaceholder() {
+        SemtypePlaceholder sp = new SemtypePlaceholder();
```
Comment (manuranga) [inline semtypes/src/test/java/org/ballerina/semtypes/SemtypeTests.java:14]: I think we'll want to directly instantieate UniformTypeBitSet , not the place holder

File: semtypes/spotbugs-exclude.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  ~  software distributed under the License is distributed on an
+  ~  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~  KIND, either express or implied.  See the License for the
+  ~  specific language governing permissions and limitations
+  ~  under the License.
+  -->
+
+<FindBugsFilter>
+    <Match>
+        <Package name="org.ballerina.semType"/>
```
Comment (manuranga) [inline semtypes/spotbugs-exclude.xml:21]: package names should be all lower case, test package and src package should be the same, this helps us to unit tests non-pubic methods

Comment (ushirask) [reply semtypes/spotbugs-exclude.xml]: fixed

File: settings.gradle [hunk 7]
```diff
@@ -113,6 +113,7 @@ include(':project-api-test-artifact:bad-sad-compiler-plugin')
 include(':project-api-test-artifact:logging-file-appender-plugin')
 //include(':ballerina-libs')
 include(':bir-spec')
+include ('semtypes')
```
Comment (manuranga) [inline settings.gradle:116]: `:` missing

Comment (ushirask) [reply settings.gradle]: fixed

**Review by manuranga — APPROVED** [review]
LGTM

File: settings.gradle [hunk 8]
```diff
@@ -113,6 +113,7 @@ include(':project-api-test-artifact:bad-sad-compiler-plugin')
 include(':project-api-test-artifact:logging-file-appender-plugin')
 //include(':ballerina-libs')
 include(':bir-spec')
+include (':semtypes')
```
Comment (rdhananjaya) [inline settings.gradle:116]:
```suggestion
include(':semtypes')
```

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**